projects/gloo/pkg/plugins/consul/eds.go

```
@@ -202,7 +241,46 @@ func buildEndpoint(namespace string, service *consulapi.CatalogService, upstream
 		address = service.Address
 	}
 
-	ep := &v1.Endpoint{
+	ipAddresses, err := getIpAddresses(address, resolver)
+	if err != nil {
+		return nil, err
+	}
+
+	var endpoints []*v1.Endpoint
+	for _, ipAddr := range ipAddresses {
+		endpoints = append(endpoints, buildEndpoint(namespace, ipAddr, service, upstreams))
+	}
+	return endpoints, nil
+}
+
+func getIpAddresses(address string, resolver DnsResolver) ([]string, error) {
+	addr := net.ParseIP(address)
+	if addr != nil {
+		// the consul service address is an IP address, no need to resolve it!
+		return []string{address}, nil
+	}
+
+	// we're assuming the consul service returned a hostname instead of an IP
+	// we need to resolve this here so EDS can be given IPs (EDS can't resolve hostnames)
+	if resolver == nil {
+		return nil, eris.Errorf("Consul service returned an address that couldn't be parsed as an IP (%s), "+
```

be careful of the impact of returning an `error` here. it means one DNS entry in the consul store will prevent EDS from working.

perhaps it is better to log error here and continue? or surface an error to the `Upstream` object somehow?

---

i had thought about this, and opted to swallow errors so EDS would continue to work. decided to log warnings and DRY some code here though https://github.com/solo-io/gloo/pull/2459/commits/8a99fb908590365a2cd9f074fc0e0ca1d3fea592